src/cascadia/TerminalApp/CommandPalette.cpp

```diff
@@ -303,20 +333,36 @@ namespace winrt::TerminalApp::implementation
     {
         if (command)
         {
-            // Close before we dispatch so that actions that open the command
-            // palette like the Tab Switcher will be able to have the last laugh.
-            _close();
-
-            const auto actionAndArgs = command.Action();
-            _dispatch.DoAction(actionAndArgs);
-
-            TraceLoggingWrite(
-                g_hTerminalAppProvider, // handle to TerminalApp tracelogging provider
-                "CommandPaletteDispatchedAction",
-                TraceLoggingDescription("Event emitted when the user selects an action in the Command Palette"),
-                TraceLoggingUInt32(_searchBox().Text().size(), "SearchTextLength", "Number of characters in the search string"),
-                TraceLoggingKeyword(MICROSOFT_KEYWORD_MEASURES),
-                TelemetryPrivacyDataTag(PDT_ProductAndServicePerformance));
+            if (command.NestedCommands().Size() > 0)
+            {
+                // If this Command had subcommands, then don't dispatch the
+                // action. Instead, display a new list of commands for the user
+                // to pick from.
+                _nestedActionStack.Append(command);
+                _updateUIForStackChange();
+            }
+            else
+            {
+                // First stash the search text length, because _close will clear this.
+                auto searchTextLength = _searchBox().Text().size();
+
+                // Close before we dispatch so that actions that open the command
+                // palette like the Tab Switcher will be able to have the last laugh.
+                _close();
+
+                const auto actionAndArgs = command.Action();
+                _dispatch.DoAction(actionAndArgs);
+
+                TraceLoggingWrite(
+                    g_hTerminalAppProvider, // handle to TerminalApp tracelogging provider
+                    "CommandPaletteDispatchedAction",
+                    TraceLoggingDescription("Event emitted when the user selects an action in the Command Palette"),
+                    TraceLoggingUInt32(searchTextLength, "SearchTextLength", "Number of characters in the search string"),
+                    TraceLoggingKeyword(MICROSOFT_KEYWORD_MEASURES),
+                    TelemetryPrivacyDataTag(PDT_ProductAndServicePerformance));
+
+                _close();
```

Comment:
unneeded `_close`? - there's one on 351

Reply:
oh no, I was supposed to get rid of one of them...